src/utils/utils.cairo

```diff
@@ -159,12 +158,16 @@ namespace Helpers {
         return (res=quotient);
     }
 
-    func compute_half_uint256{range_check_ptr}(val: felt*, i: felt, res: felt) -> (res: felt) {
-        if (i == 1) {
-            return (res=res + [val]);
+    func compute_half_uint256_from_bytes{range_check_ptr}(
+        bytes_len: felt, bytes: felt*, res: felt
+    ) -> (res: felt) {
+        if (bytes_len == 1) {
+            return (res=res + [bytes]);
         }
-        let (temp_pow) = pow(256, i - 1);
-        let (res) = compute_half_uint256(val + 1, i - 1, res + [val] * temp_pow);
+        let temp_pow = pow256_rev(16 - (bytes_len - 1));
```

Comment:
fyi if you do smth like `pow_rev(16 - i)` you can as well define `pow` and do the more natural `pow(i)`